newIDE/app/src/CommandPalette/CommandManager.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+type CmdHandler = () => void | Promise<void>;
+
+export type GlobalCommand = {|
+  displayText: string,
+  enabled: boolean,
+  handler: CmdHandler,
+|};
+
+export type UIGlobalCommand = {|
+  name: string,
+  ...GlobalCommand,
+|};
+
+export default class CommandManager {
+  globalCommands: { [string]: GlobalCommand };
```

Comment:
Should we really make the distinction between "global" commands and not global?
After all, nothing is really global (even "global variables" should have been called "game variables" ;)). 
Maybe in the future we'll allow multiple instances of the editor to run or multiple projects to be opened, in which case commands could be specified to the project.

I would recommend to go simply with:
`commands: { [string]: Command }`

Reply:
I've refactored everything to just `*Command` now (removed the "global"), since in any case adding "global" right now wasn't very relevant as that's the only kind of commands in the code currently. What should I name them, when I start with adding goto commands?